### PR TITLE
feat(tooltip): add delay option

### PR DESCRIPTION
**Pull request by astme**

### CODE REVIEW

**Comment by kevinmerckx** [comment]
@astme Please fix the unit test errors

File: src/app/ui/tooltip/tooltip.directive.ts [hunk 1]
```diff
@@ -64,9 +66,12 @@ export class TooltipDirective implements AfterViewInit, OnDestroy, EventListener
 
   handleEvent(evt: Event): void {
     if (evt.type === 'mouseenter') {
-      return this.__onMouseEnter(<MouseEvent>evt);
+      this.__timerId = setTimeout(() => {
```
Comment (kevinmerckx) [inline src/app/ui/tooltip/tooltip.directive.ts]: This timer should also be cleared onDestroy

**Comment by kevinmerckx** [comment]
@astme can you add a unit test for this delay?

File: src/app/ui/tooltip/tooltip.directive.spec.ts [hunk 2]
```diff
@@ -245,4 +244,29 @@ describe('TooltipDirective behavior', () => {
...
+      templateRef,
+      viewContainerRef,
+      'browser',
+      {}
+    );
+    directive.delay = 1000;
+    directive.ngAfterViewInit();
+    directive.handleEvent(<Event>{ type: 'mouseenter' });
+    tick(1000);
+    expect(document.body.querySelectorAll('iw-tooltip-container').length).toBe(0);
```
Comment (kevinmerckx) [inline src/app/ui/tooltip/tooltip.directive.spec.ts]: Meaning there's no iw-tooltip-container

**Comment by astme** [comment]
@kevinmerckx any other test case you can think of?

**Pull request merged**